src/error.rs

```diff
@@ -69,7 +69,7 @@ impl<'tcx> Error for EvalError<'tcx> {
             EvalError::Math(..) =>
                 "mathematical operation failed",
             EvalError::InvalidBitShiftRhs(..) =>
-                "bit shift rhs negative or not an int",
+                "bit shift rhs not an int",
```

Comment:
We shouldn't have `EvalError` variants for things the type checker should catch. If a primitive bit shift RHS isn't an integer, we can just panic.